boards/beagle/beagley_ai/doc/index.rst

```diff
@@ -0,0 +1,142 @@
+versions will also support a console over RPmsg.
+
++-----------+------------+-----------------------+
+| Interface | Controller | Driver/Component      |
++===========+============+=======================+
+| UART      | on-chip    | serial port-polling   |
+|           |            | serial port-interrupt |
++-----------+------------+-----------------------+
+
+Other hardwares features are currently not supported.
+
+The default configuration can be found in the defconfig file.
+
+Future configurations will add support for GPIO, I2C, SPI, etc.
+
+Running Zephyr
+**************
+
+The AM67A does not have a separate flash for the R5 core. Because of this
+an A53 core has to load the program for the R5 core to the right memory
+address, set the PC and start the processor.
+This can be done from Linux on the A53 core via remoteproc.
+
+This is the memory mapping from A53 to the memory usable by the R5. Note that
+the R5 core always sees its local TCMA at address 0x00000000 and its TCMB0
+at address 0x41010000. (TBD)
+
+The A53 Linux configuration allocates a region in DDR that is shared with
+the R5. The amount of the allocation can be changed in the Linux device tree.
+Note that BeagleY-AI has 4GB of DDR.
+
++-------------------+---------------+--------------+--------+
+| Region            | A53 Main Addr | MAIN R5F     | Size   |
++===================+===============+==============+========+
+| ATCM              | 0x0078400000  | 0x0000000000 | 32KB   |
++-------------------+---------------+--------------+--------+
+| BTCM              | 0x0078500000  | 0x0041010000 | 32KB   |
++-------------------+---------------+--------------+--------+
+| DDR Shared Region | 0x00A2000000  | 0x00A2000000 | 16MB   |
++-------------------+---------------+--------------+--------+
+
++-------------------+---------------+--------------+--------+
+| Region            | A53 Main Addr | MCU R5F      | Size   |
++===================+===============+==============+========+
+| ATCM              | 0x0079000000  | 0x0000000000 | 32KB   |
++-------------------+---------------+--------------+--------+
+| BTCM              | 0x0079020000  | 0x0041010000 | 32KB   |
++-------------------+---------------+--------------+--------+
+| DDR Shared Region | 0x00A1000000  | 0x00A1000000 | 16MB   |
++-------------------+---------------+--------------+--------+
+
+Steps to run the image
+----------------------
+Here is an example for the :zephyr:code-sample:`hello_world` application
+targeting the MAIN domain Cortex R5F on BeagleY-AI:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: beagley_ai/j722s/main_r5f0_0
+   :goals: build
+
+For the MCU domain Cortex R5F on BeagleY-AI:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: beagley_ai/j722s/mcu_r5f0_0
+   :goals: build
+
+To load the image:
+
+| Copy Zephyr image to the /lib/firmware/ directory.
+| ``cp build/zephyr/zephyr.elf /lib/firmware/``
```

Comment:
Maybe not required for this initial commit, but we really should include steps for getting OpenOCD+gdb working for debug. Currently, it segfaults for me.

Reply:
Yes the images will have resource tables. Until we have side-loading support the only supported way to load M4 and R5 firmware on K3 is through the Linux Remoteproc, which as you said requires a resource table. This is enabled for all K3 R5 builds by default here:

https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/ti/k3/am6x/Kconfig#L29

Debug with OpenOCD is in the works, and it should include both JTAG and self-hosted support. This will be added to the documentation when it is fully functional.